script/tool/lib/src/publish_plugin_command.dart

```diff
@@ -520,25 +519,34 @@ If running this command on CI, you can set the pub credential content in the $_p
 
   /// Returns the correct path where the pub credential is stored.
   @visibleForTesting
-  static String getCredentialPath() {
+  static String? getCredentialPath() {
     return _credentialsPath;
   }
 }
 
 /// The path in which pub expects to find its credentials file.
-final String _credentialsPath = () {
+final String? _credentialsPath = () {
```

Comment:
nits: 
maybe we can move the above logic in `_ensureValidPubCredential`:
```
    final String? credentialsPath = _credentialsPath;
    if (credentialsPath == null) {
      printError('Unable to determine pub credentials path');
      throw ToolExit(1);
    }
``` 
here, so `_credentialsPath` can return a non-null String, then `static String? getCredentialPath()` can also return a non-null String

Reply:
Good thought, done.